### PR TITLE
post create

### PR DESCRIPTION
**Pull request by mtm-heinhtetsoe**
#1 post create

### CODE REVIEW

File: app/Contracts/Dao/Post/PostDaoInterface.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Contracts\Dao\Post;
+
+interface PostDaoInterface
+{
+    public function getPostsForPostPage();
+
+    public function storePostForPostPage($request);
+}
```
Comment (mtm-zunpwintphyu) [inline app/Contracts/Dao/Post/PostDaoInterface.php]: EOL (Please click enter end of the line)

File: app/Contracts/Services/Post/PostServiceInterface.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Contracts\Services\Post;
+
+interface PostServiceInterface
+{
+    public function getPostsForPostPage();
+
+    public function storePostForPostPage($request);
```
Comment (mtm-zunpwintphyu) [inline app/Contracts/Services/Post/PostServiceInterface.php]: Please write comments before the function
eg
 /**
     * Post store for post page
     * @param [array] $request
     * @return array
  */

File: app/Contracts/Services/Post/PostServiceInterface.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+}
```
Comment (mtm-zunpwintphyu) [inline app/Contracts/Services/Post/PostServiceInterface.php]: EOL

File: app/Dao/Post/PostDao.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    public function getPostsForPostPage()
+    {   
+        return Post::all();
+    }
+
+    public function storePostForPostPage($request)
+    {
+        return Post::create($request->all());
+    }
+}
```
Comment (mtm-zunpwintphyu) [inline app/Dao/Post/PostDao.php]: EOL

File: app/Http/Controllers/PostController.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Models\Post;
+use App\Contracts\Services\Post\PostServiceInterface;
```
Comment (mtm-zunpwintphyu) [inline app/Http/Controllers/PostController.php:6]: remove [ use App\Models\Post; ]

File: app/Services/Post/PostService.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Services\Post;
+
+use App\Contracts\Dao\Post\PostDaoInterface;
+use App\Contracts\Services\Post\PostServiceInterface;
+use App\Models\Post;
+
```
Comment (mtm-zunpwintphyu) [inline app/Services/Post/PostService.php:7]: remove [use App\Models\Post;]

File: app/Services/Post/PostService.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        $request->validate([
+            'title' => 'required',
+            'description' => 'required',
+            'status' => 'required'
+        ]);
+        
+        return $this->postDao->storePostForPostPage($request);
+    }
+}
```
Comment (mtm-zunpwintphyu) [inline app/Services/Post/PostService.php]: EOL

File: database/factories/PostFactory.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    public function definition()
+    {
+        return [
+            'title' => $this->faker->word,
+            'description' => $this->faker->sentence,
+            'status' => $this->faker->word,
+            'deleted_at' => null,
+        ];
+    }
+}
```
Comment (mtm-zunpwintphyu) [inline database/factories/PostFactory.php:26]: EOL

File: database/migrations/2022_11_28_074613_create_posts_table.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('posts');
+    }
+};
```
Comment (mtm-zunpwintphyu) [inline database/migrations/2022_11_28_074613_create_posts_table.php:35]: EOL

File: database/seeders/PostSeeder.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        Post::factory()->count(10)->create();
+    }
+}
```
Comment (mtm-zunpwintphyu) [inline database/seeders/PostSeeder.php:20]: EOL

File: database/seeders/PostSeeder.php [hunk 11]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Database\Seeders;
+
+use Illuminate\Database\Console\Seeds\WithoutModelEvents;
+use Illuminate\Database\Seeder;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\DB;
+use App\Models\Post;
```
Comment (mtm-zunpwintphyu) [inline database/seeders/PostSeeder.php:6]: remove
[ 
  use Illuminate\Support\Facades\Hash;
  use Illuminate\Support\Str;
  use Illuminate\Support\Facades\DB; 
]

File: package-lock.json [hunk 12]
```diff
@@ -0,0 +1,519 @@
+{
```
Comment (mtm-zunpwintphyu) [inline package-lock.json]: [ package-lock.json ] this file doesn't need to push at git. Please hide this file

File: resources/assets/sass/post-create.scss [hunk 13]
```diff
@@ -0,0 +1,91 @@
+$background-color: #D7CCC8;
+$font-family: 'Roboto', sans-serif;
+
+@mixin btn($fontcolor: #fafafa) {
+    font-family: 'Roboto', sans-serif;
+    font-size: 15px;
```
Comment (mtm-zunpwintphyu) [inline resources/assets/sass/post-create.scss]: font-family: $font-family;

File: resources/assets/sass/post-create.scss [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    outline: 0;
+    border: 0;
+    width: 100%;
+    padding: 15px;
+    background: #ff5722;
+    color: $fontcolor;
```
Comment (mtm-zunpwintphyu) [inline resources/assets/sass/post-create.scss]: **background: #ff5722;**
background-color declares and calls it. Please do to same with another background color.

File: resources/assets/sass/post.scss [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    tr {
+        border-bottom: 5px solid $lfs-grey;
+        color: $lfs-grey;
+        td {
+            padding: 10px;
+            border: 1px solid $lfs-blue;
+        }
+    }
+}
```
Comment (mtm-zunpwintphyu) [inline resources/assets/sass/post.scss]: EOL

File: resources/js/app.js [hunk 16]
```diff
@@ -1 +1,3 @@
 import './bootstrap';
+import '../assets/sass/post.scss';
+import '../assets/sass/post-create.scss';
```
Comment (mtm-zunpwintphyu) [inline resources/js/app.js]: EOL

File: resources/views/posts/create.blade.php [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+                <legend>Create Post</legend>
+                <input type="text" name="title" placeholder="Title">
+                <input type="text" name="description" placeholder="Description">
+                <input type="text" name="status" placeholder="Status">
+                <button type="submit" class="btn">Create</button>
+            </form>
+        </div>
+    </div>
+</body>
+</html>
```
Comment (mtm-zunpwintphyu) [inline resources/views/posts/create.blade.php]: EOL

File: resources/views/posts/create.blade.php [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+</head>
+<body>
+    <div class="container">
+        <div class="form">
+            <form action="#" class="login-form">
+                <legend>Create Post</legend>
+                <input type="text" name="title" placeholder="Title">
+                <input type="text" name="description" placeholder="Description">
+                <input type="text" name="status" placeholder="Status">
+                <button type="submit" class="btn">Create</button>
```
Comment (mtm-zunpwintphyu) [inline resources/views/posts/create.blade.php]: please use a meaningful name 
class="btn" => class="createButton"

File: routes/web.php [hunk 19]
```diff
@@ -16,3 +18,6 @@
 Route::get('/', function () {
     return view('welcome');
 });
+
+Route::get('/posts', [PostController::class, 'index']);
+Route::get('/posts/create', [PostController::class, 'create']);
```
Comment (SCM-ZunPwintPhyu) [inline routes/web.php:23]: Please write like this
Route::get('/posts', 'PostController@index');

**Review by mtm-zunpwintphyu — COMMENTED** [review]
11/30/22 review

File: app/Http/Requests/PostRequest.php [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'title' => 'required',
+            'description' => 'required',
```
Comment (ThinThinMoe) [inline app/Http/Requests/PostRequest.php]: **add max length 50 for description 
add error message for max length**

File: app/Services/Post/PostService.php [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function store($request)
+    {
+        $request->validate([
+            'title' => 'required',
+            'description' => 'required',
+            'status' => 'required'
+        ]);
+
```
Comment (ThinThinMoe) [inline app/Services/Post/PostService.php]: **remove**
```
$request->validate([
            'title' => 'required',
            'description' => 'required',
            'status' => 'required'
        ]);
```

File: resources/assets/sass/post.scss [hunk 22]
```diff
@@ -0,0 +1,77 @@
+$background: #ffffff;
+$lfs-blue: #006ac6;
+$lfs-yellow: #fab700;
+$lfs-grey: #75787b;
```
Comment (ThinThinMoe) [inline resources/assets/sass/post.scss]: **Use meaningful and pronounceable variable names** 

```
$blue: #258ae2;
$white: #ffffff;
```

**Pull request merged**